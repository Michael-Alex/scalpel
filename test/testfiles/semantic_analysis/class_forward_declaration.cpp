struct s;
struct s;
struct s
{
};

namespace n
{
	struct s;
}

struct n::s
{
};

