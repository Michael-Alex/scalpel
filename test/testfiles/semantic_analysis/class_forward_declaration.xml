<namespace id="n0">
 <namespace name="n" id="n1">
  <class name="s" id="c0">
   <destructor access="public" defined="false">
   </destructor>
  </class>
 </namespace>
 <class name="s" id="c1">
  <destructor access="public" defined="false">
  </destructor>
 </class>
</namespace>
