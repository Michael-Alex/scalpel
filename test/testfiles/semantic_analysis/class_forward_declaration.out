{
  "id": "namespace-0",
  "namespaces":
  [
    {
      "id": "namespace-1",
      "name": "n",
      "classes":
      [
        {
          "id": "class-0",
          "name": "s",
          "destructor":
          {
            "access": "public"
          }
        }
      ]
    }
  ],
  "classes":
  [
    {
      "id": "class-1",
      "name": "s",
      "destructor":
      {
        "access": "public"
      }
    }
  ]
}
